test/arguments.ts

```diff
@@ -621,7 +618,6 @@ test('options have url even if some are invalid - got.extend', async t => {
 	});
 
 	await t.throwsAsync(instance('https://example.com', {
-		// @ts-expect-error
```

Comment:
I think this illustrates why this change is not an improvement. It makes the types looser.